### ISSUE TITLE
Problems with Docker and LDAP

### ISSUE THREAD

**Issue by moelfus** [Net-ng/kansha#236]
Hi, 

I wanted to test Kansha with Docker and a LDAP connection. Therefore I cloned the git project and created a docker image from it. To get the LDAP connection running I added the line 

`RUN /opt/stackless/bin/easy_install kansha[ldap]`

to the Dockerfile. I also altered the configuration file (conf/kansha.cfg) as follows:

```
[authentication]
[[dblogin]]
activated = off
moderator = kansha@net-ng.com

[[ldaplogin]]
activated = on
server = server.example
users_base_dn = "ou=People,dc=server,dc=example"
cls = kansha.authentication.ldap.ldap_auth:NngLDAPAuth 
```

While the image is build I got some warnings  (... warning: build_py: file Lib/ldap.py (for module ldap) not found ...  warning: build_py: file Lib/ldap/schema.py (for module ldap.schema) not found)  but ignored them as mentioned in the python-ldap FAQ. When I run the image the webpage shows an internal error message and the logfile prints:

`2017-01-18 15:26:39,941 - nagare.application.kansha - ERROR - 'module' object has no attribute                                'NngLDAPAuth'
`
Might this be a bug or did I something wrong?

Anyways, cool program!

**Comment by vrialland** [comment]
Hi,

there has been some renaming in the config parameters that have not been reflected in the docs yet:
- `server` has been renamed to `host`
- `cls` has been renamed to `schema`

Related code is located [here](https://github.com/Net-ng/kansha/blob/master/kansha/authentication/ldap/forms.py#L23).
So you have to alter your configuration file to change these.

Moreover, the schema param default is `kansha.authentication.ldap:NngLDAPAuth` which is not accessible, it should be `kansha.authentication.ldap.ldap_auth:NngLDAPAuth` but you have already found out.

Within the LDAP module, you have 3 classes: `LDAPAuth`, a base class to connect to an LDAP server, `ADLDAPAuth` for Active Directory and `NngLDAPAuth` which is used internally at Net-ng to load data from our LDAP server (see [this](https://github.com/Net-ng/kansha/blob/master/kansha/authentication/ldap/ldap_auth.py#L86) for fields loading).
As LDAP storage may vary a lot between servers, it may work, or you may have to create your own to load your data properly.


**tl;dr:**

Edit the '[[ldaplogin]]' to look like this:

```ini
[[ldaplogin]]
activated = on
host = server.example
users_base_dn = "ou=People,dc=server,dc=example"
schema = kansha.authentication.ldap.ldap_auth:NngLDAPAuth 
```

If the data fields in your LDAP server do not match those in `NngLDAPAuth` you will have to subclass `LDAPAuth` to fit your needs.

**Comment by rtxm** [comment]
`NngLDAPAuth ` expects an InetOrgPerson schema.

**Comment by moelfus** [comment]
Thanks for your help. Now it is working. When you update the documentation you could add that the LDAP field "displayName" and "mail" must be set for the user, otherwise an error is thrown. This is interesting when using LAM since the "displayName" field is not shown there by default.
